hosted-cluster-config-operator/main.go

```diff
@@ -169,17 +172,37 @@ func (o *HostedClusterConfigOperator) Run(ctx context.Context) error {
 		"release":    o.ReleaseVersion,
 		"kubernetes": o.KubernetesVersion,
 	}
-	cfg := operator.NewHostedClusterConfigOperatorConfig(
-		o.TargetKubeconfig,
-		o.Namespace,
-		o.HostedControlPlaneName,
-		o.initialCA,
-		versions,
-		o.Controllers,
-		controllerFuncs,
-		o.enableCIDebugOutput,
-		hyperv1.PlatformType(o.platformType),
-		o.clusterSignerCA,
-	)
-	return cfg.Start(ctx)
+	cfg := operator.CfgFromFile(o.TargetKubeconfig)
+	cpConfig := ctrl.GetConfigOrDie()
+	mgr := operator.Mgr(cfg, cpConfig, o.Namespace)
+	cpCluster, err := cluster.New(cpConfig, func(opt *cluster.Options) {
+		opt.Namespace = o.Namespace
+		opt.Scheme = api.Scheme
+	})
+	if err != nil {
+		panic(fmt.Sprintf("Cannot create control plane cluster: %v", err))
```

Comment:
We can actually return an error here and below now instead of `panic` ing :)

Reply:
fixed